src/cpp/src/llm_pipeline.cpp

```diff
@@ -264,31 +302,29 @@ class LLMPipeline::LLMPipelineImpl {
         return tpl.RenderAsString(params).value();
     }
 
-    std::string apply_chat_template(std::string prompt, std::string role = "user") const {
+    std::string apply_chat_template(const ChatHistory& history, bool add_generation_prompt) const {
```

Comment:
It seems you don't need `std::string apply_chat_template(const std::vector<std::pair<std::string, std::string>>& prompts) const` overload anymore

Reply:
you are right, removed that